### PR TITLE
Added the Kubernetes scheduler to the component table

### PR DESCRIPTION
**Pull request by bergerhoffer**
Preview: https://deploy-preview-32183--osdocs.netlify.app/openshift-enterprise/latest/architecture/control-plane.html#defining-masters_control-plane

### CODE REVIEW

**Comment by bergerhoffer** [comment]
@soltysh Can you review this update to add the Kubernetes scheduler to the control plane component list? I'll backport the change to 4.5+. I also removed HAProxy services from the sentence above the table - let me know if that is correct or not.

Preview: https://deploy-preview-32183--osdocs.netlify.app/openshift-enterprise/latest/architecture/control-plane.html#defining-masters_control-plane

File: modules/architecture-machine-roles.adoc [hunk 1]
```diff
@@ -42,18 +42,23 @@ Services that fall under the Kubernetes category on the master include the Kuber
...
 //etcd can be optionally configured for high availability, typically deployed with 2n+1 peer services.
+
 |Kubernetes controller manager
 |The Kubernetes controller manager watches etcd for changes to objects such as
 replication, namespace, and service account controller objects, and then uses the
 API to enforce the specified state. Several such processes create a cluster with
 one active leader at a time.
+
+|Kubernetes scheduler
+|The Kubernetes scheduler watches for newly created pods and selects the best node for the pod to run on.
```
Comment (skrthomas) [inline modules/architecture-machine-roles.adoc:58]: consider "and selects the best node to run the pod." to de-clutter some prepositions.

**Review by skrthomas — COMMENTED** [review]
Just one small suggestion, otherwise LGTM!

File: modules/architecture-machine-roles.adoc [hunk 2]
```diff
@@ -33,7 +33,7 @@ In a Kubernetes cluster, the master nodes run services that are required to cont
 Exactly three master nodes must be used for all production deployments.
 ====
 
-Services that fall under the Kubernetes category on the master include the Kubernetes API server, etcd, Kubernetes controller manager, and HAProxy services.
```
Comment (bergerhoffer) [inline modules/architecture-machine-roles.adoc:36]: @knobunc Can you confirm whether or not it's appropriate to remove "HAProxy services" from this sentence here?

We had HAProxy services as an entry in the table in 3.11 [1] but it's not in the table for 4.x [2].

[1] https://docs.openshift.com/container-platform/3.11/architecture/infrastructure_components/kubernetes_infrastructure.html#master
[2] https://docs.openshift.com/container-platform/4.7/architecture/control-plane.html#defining-masters_control-plane

**Review by soltysh — APPROVED** [review]
/lgtm

File: modules/architecture-machine-roles.adoc (hunk 2 above)
Comment (bergerhoffer) [reply modules/architecture-machine-roles.adoc:36]: Per slack conversation with @knobunc, HAProxy services should be removed from this sentence.

File: modules/architecture-machine-roles.adoc [hunk 3]
```diff
@@ -42,18 +42,23 @@ Services that fall under the Kubernetes category on the master include the Kuber
...
 replication, namespace, and service account controller objects, and then uses the
 API to enforce the specified state. Several such processes create a cluster with
 one active leader at a time.
+
+|Kubernetes scheduler
+|The Kubernetes scheduler watches for newly created pods and selects the best node to host the pod.
 |===
 
-There are also OpenShift services that run on the control plane, which include the OpenShift API server, OpenShift controller manager, and OAuth API server.
+There are also OpenShift services that run on the control plane, which include the OpenShift API server, OpenShift controller manager, OpenShift OAuth API server, and OpenShift OAuth server.
```
Comment (kasturinarra) [inline modules/architecture-machine-roles.adoc:61]: @bergerhoffer why do we have these changed ? In the other PR i do not see changes here.

File: modules/architecture-machine-roles.adoc [hunk 4]
```diff
@@ -42,18 +42,23 @@ Services that fall under the Kubernetes category on the master include the Kuber
...
 //etcd can be optionally configured for high availability, typically deployed with 2n+1 peer services.
+
 |Kubernetes controller manager
 |The Kubernetes controller manager watches etcd for changes to objects such as
 replication, namespace, and service account controller objects, and then uses the
 API to enforce the specified state. Several such processes create a cluster with
 one active leader at a time.
+
+|Kubernetes scheduler
+|The Kubernetes scheduler watches for newly created pods and selects the best node to host the pod.
```
Comment (kasturinarra) [inline modules/architecture-machine-roles.adoc:58]: same comments as https://github.com/openshift/openshift-docs/pull/32184/files

File: modules/architecture-machine-roles.adoc (hunk 3 above)
Comment (bergerhoffer) [reply modules/architecture-machine-roles.adoc:61]: Right. In 4.5, we the arch doc only had a table that listed the Kubernetes components that run on the control plane. In 4.6, we added another table that lists the OpenShift components. So this adjustment here doesn't apply to 4.5 - which is the reason that I opened a separate PR (#32184) for 4.5.

**Pull request merged**